### ISSUE TITLE
Add functions to map an array into an object

### ISSUE THREAD

**Issue by bruceg** [vectordotdev/vrl#1157]
When manipulating data that has been parsed from CSV or other columnar data sources, it is useful to be able to map those columns into an object with named fields. I propose adding two functions to accomplish this, mirroring the approach in Python: `object` and `zip`:

* `zip` takes a number of arrays as input and produces a single array as output containing a tuple of elements from each array. For example `zip([1,2,3],[4,5,6],[7,8,9])` returns `[[1,4,7],[2,5,8],[3,6,9]]`.
* `object` takes takes a zipped array as above with (key, value) pairs and produces an object as output. For example `object([["foo":1],["bar":2]])` returns `{"foo":1,"bar":2}`.

**Comment by jszwedko** [comment]
I'd suggest a different name for the function that takes an array of "tuples" than `object`, which [already exists](https://vector.dev/docs/reference/vrl/functions/#object). Maybe `from_pairs`? That's what lodash calls it. Maybe there are other examples from other languages or libraries we could use for inspiration.

**Comment by bruceg** [comment]
I missed that `object` was already a function, and I'm on board with something like `from_pairs`, but isn't that a little ambiguous as to what the _output_ of such a function would be?

**Comment by jszwedko** [comment]
True, could call it `object_from_pairs` to be explicit. Granted we have other functions already where the return type isn't clear just from the name itself (e.g. [from_unix_timestamp](https://vector.dev/docs/reference/vrl/functions/#from_unix_timestamp)).
